src/idpi/grib_decoder.py

```diff
@@ -333,13 +326,17 @@ def load(
         """
         params = set()
         for desc in descriptors:
-            params |= set(desc.input_fields)
+            params |= set([frozendict(data_desc) for data_desc in desc.input_fields])
 
         return self.load_fields(params, extract_pv=extract_pv)
 
+    def load_fieldnames(self, params: list[str], extract_pv: str | None = None):
+        desc = ProductDescriptor(input_fields=[{"param": param} for param in params])
+        return self.load([desc], extract_pv=extract_pv)
+
     def load_fields(
         self,
-        params: typing.Iterable[str],
+        params: typing.Iterable[dict],
```

Comment:
Namedtuple or dataclass would also be a better fit for this interface. Now, you would need to go dig in the earthkit data docs to figure out what are admissible keys in the dict.